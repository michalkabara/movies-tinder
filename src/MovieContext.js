import React, {useState, useEffect} from 'react'
import axios from 'axios'

const MovieContext = React.createContext()

const MovieContextProvider = ({children}) => {

    const [movieList, setMovieList] = useState([])

    useEffect(() => {
        const fetchData = async () => {
            const result = await axios("http://localhost:3004/movies")
            setMovieList(result.data)
    }
    fetchData()
    },[])

    const AcceptMovie = (id, title, imageURL,summary,rating) => {
        setMovieList(prevItems => prevItems.filter(item => item.id !== id))

        axios.post(`/recommendations/${id}/accept`,{
            acceptedMovies: {
                id,
                title,
                imageURL,
                summary,
                rating
            }
        })
          .then(function (response) {
            console.log(response);
          })
          .catch(function (error) {
            console.log(error);
          });
    }

    const RejectMovie = (id, title, imageURL,summary,rating) => {
        setMovieList(prevItems => prevItems.filter(item => item.id !== id))

        axios.post(`/recommendations/${id}/reject`,{
            rejecteddMovies: {
                id,
                title,
                imageURL,
                summary,
                rating
            }
        })
          .then(function (response) {
            console.log(response);
          })
          .catch(function (error) {
            console.log(error);
          });

    }


    return(
        <MovieContext.Provider value={{movieList, AcceptMovie, RejectMovie}}>
            {children}
        </MovieContext.Provider>
    )
}

export {MovieContext, MovieContextProvider}