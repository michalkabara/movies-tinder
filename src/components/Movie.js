import React from "react";

const Movie = (props) =>{
        return(
            <div className="movieContainer">
                <h2>{props.movie.title}</h2>
                <img alt="movie poster" className="moviePoster" src={props.movie.poster}></img>
                <p>{props.movie.summary}</p>
                <p>({props.movie.rating}/10)</p>
                <p>{props.movie.key}</p>
            </div>
        )
    }
export default Movie