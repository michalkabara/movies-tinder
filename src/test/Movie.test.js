// import React from 'react';
// import Movie from '../components/Movie'
// import renderer from 'react-test-renderer';
// import {render, cleanup} from '@testing-library/react'
// import '@testing-library/jest-dom/extend-expect'

// afterEach(cleanup)
// it('matches snapshot', () => {
//     const tree = renderer.create(<Movie />).toJSON()
//     expect(tree).toMatchSnapshot()
// });

import {mount} from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'
import Movie from '../components/Movie'
import {MovieContextProvider} from '../MovieContext'

test("matches snapshot", () => {
    const wrapper = mount(
        <MovieContextProvider>
    <Movie />
    </MovieContextProvider>)
    expect(toJSON(wrapper).toMatchSnapshot())
})