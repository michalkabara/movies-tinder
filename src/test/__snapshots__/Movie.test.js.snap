// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<MovieContextProvider>
  <Movie>
    <div>
      <button
        className="acceptButton"
        disabled={true}
        onClick={[Function]}
      >
        Accept
      </button>
      <button
        className="rejectButton"
        disabled={true}
        onClick={[Function]}
      >
        Reject
      </button>
      <button
        className="tryAgainButton"
        onClick={[Function]}
      >
        Try again :)
      </button>
    </div>
  </Movie>
</MovieContextProvider>
`;
